cupy/random/_generator_api.pyx

```diff
@@ -495,6 +498,30 @@ class Generator:
         # omitted args.
         return (<object>y).astype(dtype, copy=False)
 
+    def binomial(self, n, p, size=None):
+        """binomial"""
+        cdef ndarray y
+        cdef ndarray n_arr
+        cdef ndarray p_arr
+
+        n = cupy.asarray(n)
+        p = cupy.asarray(p)
+
+        if size is None:
+            size = cupy.broadcast(n, p).shape
+
+        y = ndarray(size if size is not None else (), numpy.int64)
+
```

Comment:
need to add several checks as in https://github.com/cupy/cupy/pull/5182

Reply:
Should I also add checks for the input domain like for `p >= 0 and p<=1`?

Reply:
This check is missing in the distributions.

Reply:
that check involves device synchronization so I think it is ok not do it, but leave it as a comment

also p can be an array so check the above PR to deal with it accordingly :)